pkg/kv/store_message.go

```diff
@@ -51,62 +51,133 @@ func (s *StoreMessage) DeleteMsg(ctx context.Context, path string) error {
 	return s.Delete(ctx, []byte(path))
 }
 
-// Scan returns a prefix iterator which returns entries in deserialized format. Scan in store message implementation is msg specific.
-//   Therefore, the prefix given should limit the range of the data to those that can be deserialized by the proto message type.
-func (s *StoreMessage) Scan(ctx context.Context, msgType protoreflect.MessageType, prefix string) (*MessageIterator, error) {
-	return NewMessageIterator(ctx, s.Store, msgType, prefix)
-}
-
-type MessageIterator struct {
-	itr     EntriesIterator
-	msgType protoreflect.MessageType
-	err     error
+func (s *StoreMessage) Scan(ctx context.Context, prefix string) (*PrimaryIterator, error) {
+	return NewPrimaryIterator(ctx, s.Store, prefix)
 }
 
 type MessageEntry struct {
 	Key   string
 	Value protoreflect.ProtoMessage
 }
 
-func NewMessageIterator(ctx context.Context, store Store, msgType protoreflect.MessageType, prefix string) (*MessageIterator, error) {
+func NewMessageEntry(msg protoreflect.MessageType) *MessageEntry {
+	return &MessageEntry{
```

Comment:
It's strange that a `NewMessageEntry` returns a  `MessageEntry` with uninitialzed fields.

Reply:
The struct is no longer relevant after the refactoring